### ISSUE TITLE
Create a snapcraft.yaml file for the DSS

### ISSUE THREAD

**Issue by DnPlas** [canonical/data-science-stack#25]
### Why it needs to get done

A `snapcraft.yaml` file must be created defining all what is needed for the DSS to be installed, started, removed, etc.

### What needs to get done

To write a `snapcraft.yaml`.  The actual content of the snap depends on the exploration task #23, this field will be updated once we have worked on it.

Reference:
* https://snapcraft.io/first-snap/python/linux/package
* https://snapcraft.io/build

### When is the task considered done

TODO: depends on #23 to clearly define what are the contents of the `snapcraft.yaml` file and deliverables of this task.

**Comment by DnPlas** [comment]
Following the docs for [creating a snap](https://snapcraft.io/docs/creating-a-snap), here are some details about the implementation.

### [Snapcraft checklist](https://snapcraft.io/docs/snapcraft-checklist)

1. Language: python3
2. Toolkits and desktop support: not needed
3. System integration:
* Access to the KUBECONFIG file, usually placed in `~/.kube/config`. The only thing that could be a problem with this is that snap interfaces usually access non-hidden files. To workaround this we may need to place a non-hidden file in the user's home directory. The [home interface](https://snapcraft.io/docs/home-interface) will help out with this.

**Comment by DnPlas** [comment]
List of nice to have:

* [tab completion](https://snapcraft.io/docs/tab-completion)

**Comment by DnPlas** [comment]
Design choices:

1. We need to create our own plugin for accessing hidden files in `$HOME/.dss/config` to be able to get the KUBECONFIG from there. Users will have to save it there always.

```
plugs:
  dot-dss-config:
    interface: personal-files
    read:
      - $HOME/.dss/config
 ```

This requires uses to connect manually with `sudo snap connect dss:dot-dss-config`. This is also something we'll have to document.

2. The snap version is picked up automatically from Github tags. This will help us when automating the release process.
3. We'll use both the `home` and the `network` interfaces. The former allows us to access files in the home directory and the latter allows `dss` to access the microk8s API.
